client/client.go

```diff
@@ -218,14 +234,41 @@ func (c Client) RoleTargetsPath(roleName string, snapshotMeta data.Files, consis
 		}
 		roleName = path.Join(
 			dir,
-			fmt.Sprintf("%s.%s.json", roleMeta.Hashes["sha256"], roleName),
+			fmt.Sprintf("%s.%s.json", hasheSha256, roleName),
 		)
 	}
 	return roleName, nil
 }
 
-func (c Client) TargetMeta(path string) *data.FileMeta {
-	return c.local.FindTarget(path)
+// TargetMeta ensures the repo is up to date, downloading the minimum
+// necessary metadata files
+func (c Client) TargetMeta(path string) (*data.FileMeta, error) {
+	c.Update()
+	var meta *data.FileMeta
+
+	// FIFO list of targets delegations to inspect for target
+	roles := []string{data.ValidRoles["targets"]}
+	for ; len(roles) > 0; role := roles[0] {
+		roles = roles[1:]
+		// Download the target role file if necessary
+		err = c.downloadTargets(role)
+		if err != nil {
+			// as long as we find a valid target somewhere we're happy.
+			// continue and search other delegated roles if any
+			continue
+		}
+
+		meta = c.local.TargetMeta(role, path)
+		if meta != nil {
+			// we found the target!
+			break
+		}
+		delegations := c.local.TargetDelegations(role, path, pathHex)
+		for _, d := range delegations {
+			roles = append(roles, d.Name)
+		}
+	}
+	return meta, nil
```

Comment:
Is it intended that this function returns a `nil` meta if nothing is found? That seems to be the current logic.